### PR TITLE
Text Proposal for Data Protection

### PR DESCRIPTION
**Pull request by hannestschofenig**
Here is my argument for making this text change: 

I believe that the architecture document should allow for different solutions that follow certain assumptions. Of course, the architecture puts constraints on the solution space but, as outlined in various sections in the draft (such as Figure 5) there are a few options. 

The text I have modified with this PR focuses on a specific solution, namely one that follows the TEEP protocol. I do, however, believe that there will be other designs in the future that will be in the scope of the TEEP architecture document but make different design decisions. 

I believe that there is no reason why we wouldn't want to allow them as well.

### CODE REVIEW

File: draft-ietf-teep-architecture.md
Comment (dthaler) [inline draft-ietf-teep-architecture.md]: This wording implies that only TLS is legal under the TEEP protocol, which is not the case.  It's up to each transport protocol to specify.

```suggestion
the transport protocol and any security mechanism associated with it (e.g., the Transport Layer Security protocol) under 
```

Comment (mingpeiwk) [reply draft-ietf-teep-architecture.md]: The change looks good to me.

**Review by mingpeiwk — APPROVED** [review]
Agree the text change and the proposed line.

**Review by dthaler — APPROVED** [review]
Approving after merging in my wordsmithing per Ming's agreement.

**Pull request merged**
